sdk/rust/oak/src/handle.rs

```diff
@@ -0,0 +1,102 @@
+include!(concat!(env!("OUT_DIR"), "/oak.handle.rs"));
+
+use crate::OakError;
+
+pub type Handle = oak_abi::Handle;
+
+pub trait HandleVisit {
+    fn visit<F: FnMut(&mut Handle)>(&mut self, visitor: F) -> F;
+}
+
+macro_rules! handle_visit_blanket_impl {
+    ($($t:ty),+) => {
+        $(
+            impl HandleVisit for $t {
+                fn visit<F: FnMut(&mut Handle)>(&mut self, visitor: F) -> F {
+                    visitor
+                }
+            }
+        )+
+    };
+}
+handle_visit_blanket_impl!((), f64, f32, i32, i64, u32, u64, bool, String, Vec<u8>);
+handle_visit_blanket_impl!(
+    oak_abi::proto::oak::encap::GrpcResponse,
+    oak_abi::proto::oak::encap::GrpcRequest,
+    oak_abi::proto::oak::log::LogMessage,
+    oak_abi::proto::oak::application::ConfigMap
+);
+
+pub fn extract_handles<T: HandleVisit>(msg: &mut T) -> Vec<Handle> {
+    let mut handles = Vec::new();
+    msg.visit(|handle: &mut Handle| {
+        handles.push(*handle);
+        *handle = 0;
```

Comment:
Comment why this is the case. Also use INVALID_HANDLE or some other constant if possible.

Reply:
Done.